orca-core/src/test/java/com/netflix/spinnaker/orca/pipeline/expressions/functions/ServiceKeyExpressionFunctionProviderTest.java

```diff
@@ -0,0 +1,128 @@
+    Map<String, Object> contextWithServiceKey = new HashMap<>();
+    contextWithServiceKey.put("parentStage", "parent-stage");
+    contextWithServiceKey.put("account", "account-name");
+    contextWithServiceKey.put("credentials", "my-account");
+    contextWithServiceKey.put("region", "org > space");
+    contextWithServiceKey.put("kato.tasks", Collections.singletonList(katoTaskMapBuilderWithResults.build()));
+    Map<String, Object> contextWithoutServiceKey = new HashMap<>();
+    contextWithoutServiceKey.put("parentStage", "parent-stage");
+    contextWithoutServiceKey.put("account", "account-name");
+    contextWithoutServiceKey.put("credentials", "my-account");
+    contextWithoutServiceKey.put("region", "org > space");
+    contextWithoutServiceKey.put("kato.tasks", Collections.singletonList(katoTaskMapBuilderWithoutResults.build()));
+    Map<String, Object> contextWithRunningTask = new HashMap<>();
+    contextWithRunningTask.put("parentStage", "parent-stage");
+    contextWithRunningTask.put("account", "account-name");
+    contextWithRunningTask.put("credentials", "my-account");
+    contextWithRunningTask.put("region", "org > space");
+    contextWithRunningTask.put("kato.tasks", Collections.singletonList(katoTaskMapBuilderRunning.build()));
+
+    Stage stage1 = new Stage(new Execution(PIPELINE, "orca"), "createServiceKey", "stage-name-1", contextWithServiceKey);
+    stage1.setStatus(SUCCEEDED);
+    Stage stage2 = new Stage(new Execution(PIPELINE, "orca"), "deployService", "stage-name-2", contextWithoutServiceKey);
+    stage2.setStatus(SUCCEEDED);
+    Stage stage3 = new Stage(new Execution(PIPELINE, "orca"), "createServiceKey", "stage-name-3", contextWithoutServiceKey);
+    stage3.setStatus(SUCCEEDED);
+    Stage stage4 = new Stage(new Execution(PIPELINE, "orca"), "createServiceKey", "stage-name-4", contextWithRunningTask);
+    stage4.setStatus(RUNNING);
+
+    execution.getStages().add(stage1);
+    execution.getStages().add(stage2);
+    execution.getStages().add(stage3);
+    execution.getStages().add(stage4);
+
+    Map<String, Object> expectedServiceKey = new ImmutableMap.Builder<String, Object>()
+      .put("username", user)
+      .put("password", password)
+      .put("url", url)
+      .build();
+
+    Map<String, Object> resultServiceKey = ServiceKeyExpressionFunctionProvider.serviceKey(execution, "stage-name-1");
+
+    assertThat(resultServiceKey).isEqualTo(expectedServiceKey);
+  }
+
+  @Test
+  void serviceKeyShouldReturnEmptyMapForNoResult() {
+    Map<String, Object> katoTaskMapWithoutResults = new ImmutableMap.Builder<String, Object>()
+      .put("id", "task-id")
+      .put("status", SUCCEEDED)
+      .put("history", Collections.emptyList())
+      .put("resultObjects", Collections.emptyList())
+      .build();
+    Execution execution = new Execution(PIPELINE, "stage-name-1", "application-name");
+    Map<String, Object> contextWithoutServiceKey = new HashMap<>();
```

Comment:
```
    Map<String, Object> contextWithoutServiceKey = new HashMap<>();
    contextWithoutServiceKey.put("parentStage", "parent-stage");
    contextWithoutServiceKey.put("account", "account-name");
    contextWithoutServiceKey.put("credentials", "my-account");
    contextWithoutServiceKey.put("region", "org > space");
```
is repeated a bunch of times, can make a helper function to make test more readable